src/tester.rs

```diff
@@ -470,150 +484,163 @@ fn run_tests(
+                        failure.status = Some("Exit was not due to signal".to_owned());
                     }
-
-                    if !pass_stdout || failure.stdout.is_none() {
-                        failure.stdout = Some(stdout);
+                    Status::Int(_) => {
+                        failure.status =
+                            Some(status.code().map(|x| x.to_string()).unwrap_or_else(|| {
+                                format!("Exited due to signal: {}", status.signal().unwrap())
+                                    .to_owned()
+                            }))
                     }
-
-                    // If a sub-test failed, bail out immediately, otherwise subsequent sub-tests
-                    // will overwrite the failure output!
-                    break;
                 }
+            }
 
-                // If a command failed, and we weren't expecting it to, bail out immediately.
-                if !status.success() && meant_to_error {
-                    break;
-                }
+            if !pass_stderr || failure.stderr.is_none() {
+                failure.stderr = Some(stderr);
             }
 
-            {
-                // Grab a lock on stderr so that we can avoid the possibility of lines blurring
-                // together in confusing ways.
-                let stderr = StandardStream::stderr(ColorChoice::Always);
-                let mut handle = stderr.lock();
-                if inner.test_threads > 1 {
-                    handle
-                        .write_all(&format!("\ntest lang_tests::{} ... ", test_fname).as_bytes())
-                        .ok();
-                }
-                if failure
-                    != (TestFailure {
-                        status: None,
-                        stderr: None,
-                        stdout: None,
-                    })
-                {
-                    let mut failures = failures.lock().unwrap();
-                    failures.push((test_fname, failure));
-                    handle
-                        .set_color(ColorSpec::new().set_fg(Some(Color::Red)))
-                        .ok();
-                    handle.write_all(b"FAILED").ok();
-                    handle.reset().ok();
-                } else {
-                    handle
-                        .set_color(ColorSpec::new().set_fg(Some(Color::Green)))
-                        .ok();
-                    handle.write_all(b"ok").ok();
-                    handle.reset().ok();
-                }
+            if !pass_stdout || failure.stdout.is_none() {
+                failure.stdout = Some(stdout);
             }
-        });
+
+            // If a sub-test failed, bail out immediately, otherwise subsequent sub-tests
+            // will overwrite the failure output!
+            break;
+        }
+
+        // If a command failed, and we weren't expecting it to, bail out immediately.
+        if !status.success() && meant_to_error {
```

Comment:
Is this right? The comment seems to contradict the conditional.

Reply:
I think it obeys it? I didn't change this bit anyway (other than moving the code, which I admit makes it look like a change).

Reply:
Fair enough. Lets leave it alone if there was no functional change!